portality/formcontext/formcontext.py

```diff
@@ -871,8 +871,16 @@ def _set_choices(self):
                 # Get the full tuple of application status for the current source
                 [full_current_status] = filter(lambda x: self.source.application_status in x, editor_choices)
 
-                # Only show options forward in the editorial pipeline
-                self.form.application_status.choices = editor_choices[editor_choices.index(full_current_status):]
+                # Show options forward in the editorial pipeline
+                forward_choices = editor_choices[editor_choices.index(full_current_status):]
+
+                # But allow an exception: reverts to 'in progress' from 'completed'
+                if self.source.application_status == 'completed':
```

Comment:
Yeah, same comment as above - let's encapsulate the choices the user had in the Choices module as much as possible.